(stoa0022a)

AlisonBabeu opened the pull request:
Converted stoa0040a (Pseudo-Augustine) to stoa0022a *(Ambrosiaster), re long ago issue #305.
While had created catalog_data for this edition at the time had no way to change the data.
Renumbered file and also combined the two-only one work.

lcerrato commented:
@AlisonBabeu 
It looks like you added books to stoa0022a.stoa001.opp-lat1.xml, is that correct? There are some oddly named divs in this file.

AlisonBabeu commented:
hi @lcerrato  those comments were just for myself as I was combining two XML files I had planned to do further updating.
I hadn't asked for your review yet as I always do because I wanted to see if it worked and if I could figure out fixing it myself before bothering you! :)

 This is all one work but there were two XML files, one had the main body and the other had what I guess for lack of a better term are appendices, its very complicated. When it first failed I tried changing the names of the sections, the previous names of the two new sections were originally "neu" and "old" or something like that. I spent the better part of an hour trying to think about how to name the next two divs or best combine this file.

lcerrato commented:
@AlisonBabeu 
Let me know if you want to revisit the editing. 
For instance, you can name the divs `appendix1` or `appendix_A` or`app_1` etc. 

Something that reflects that these are appendices is better than making them books 2 and 3, if they are not books 2 and 3. And would be much better than `neu` or `old`. (They could be `appendix_new` etc.)

You cannot have a `.` in an n attribute. That was part of the initial issue.

AlisonBabeu commented:
hi @lcerrato so I changed the divs back to appendix1 and 2 and added a change log but at this point I'm not sure how to fix it.  I see in the Travis message for the broken build  Forbidden characters found: 'appendix2.29 '" Forgive my ignorance but I can't find what is mean by this in the file anywhere. I know this exact string is not in TEI-XMl file so what am I missing.

lcerrato commented:
@AlisonBabeu 
It won't pass until the refsDecl reflects the correct file structure.

So, first, you need to enclose the original file in a book level div. You've added on two new books but the initial work now needs to be in the same level. So you need a `n="main"` or `n="0"` or something to designate the text you had there before.

Then the refsDecl needs to add this level.

lcerrato commented:
@AlisonBabeu 
You also have checked in a /.directory file that should be deleted.

lcerrato commented:
@AlisonBabeu 
`Forbidden characters found:
[215]()	stoa0022a.stoa001.opp-lat1.xml	'appendix2.29 '`

This means there is something in the markup that is not allowed.  In this case, there is a space after `29` as `<div n="29 "`

AlisonBabeu commented:
So @lcerrato I updated the refsDecl and enclosed the first text in a book level div. This has graduated me to a new set of errors:

 Unique nodes found by XPath 
   Word Counting        
Passage level parsing    
  Empty References

At this point I must admit I'm stumped.

lcerrato commented:
@AlisonBabeu 
Because the count is only picking up book and chapter (note the Nodes are 3;246 when you want three digits there), the problem is in the section level. Although I no longer use this refsDecl format, I think that is ok (that's always a place to look).

There are probably chapters that do not have sections.

lcerrato commented:
@AlisonBabeu 

For instance, 

```
<div n="appendix1" subtype="book" type="textpart">
 <head>
  <title type="main">QVAESTIONES [SANCTI AUGUSTINI] DE UETERI ET NOUO TESTAMENTO. </title>
 </head> 
<div n="1a" subtype="chapter" type="textpart">
<ab> <title>I. </title> </ab>
          <p>I huius recensionis = I recensionis quaestionum numero CXXVII. 
</p>
</div>
```

here https://archive.org/details/corpusscriptoru16wiengoog/page/419/mode/2up?view=theater

That is not a chapter. There is a "book/appendix" title, then that's just an explanation of the title, so something like a subtitle. But because that has no section, it's not going to pass. 

On the top of this page https://archive.org/details/corpusscriptoru16wiengoog/page/421/mode/2up?view=theater

there is 
**III—XII = II-XI
XIII = XXXV
XIIII-XXXVI = XII-XXXIIII**

which has also been encoded as chapters. That's probably more of a notation, just telling the users that there are no differences here from the main text in these sections. So it's debatable that that would be made "chapters" for markup purposes. It's a can of worms, really. 

I would guess that's what's breaking this.

AlisonBabeu commented:
@lcerrato I give up at this point in all honesty, I'm going to split the files back up I think and just make them two works, rather than try and fight this out. I didn't change any of the encoding that you are reference, just added letters into the divs to avoid the duplicate node problem. Thanks for all the time you've spent.

lcerrato commented:
@AlisonBabeu 
Give me a chance to take a closer look — I can't test anything until I pull it all offline.

I didn't think you changed that encoding (it was there). I was just looking at the original files to see if they passed like this.

lcerrato commented:
@AlisonBabeu 
I think it's just the refsDecl. The old one from the deleted file is working.

AlisonBabeu reviewed (commented):
Thank you thank you for finding and fixing this @lcerrato
